src/guiapi/src/window_menu.cpp

```diff
@@ -80,33 +83,37 @@ IWindowMenuItem *window_menu_t::GetActiveItem() {
     return GetItem(index);
 }
 
-bool window_menu_t::moveToNextVisibleItem(int steps) {
-    if (steps == 0)
+bool window_menu_t::moveToNextVisibleItem() {
+    if (moveIndex == 0)
         return true;
-    int dir = SIGN1(steps); /// direction of movement (+/- 1)
-    steps = ABS(steps);
+    int dir = SIGN1(moveIndex); /// direction of movement (+/- 1)
 
     IWindowMenuItem *item;
     int moved; // number of positions moved
-    for (; steps > 0; --steps) {
+    for (; moveIndex != 0; moveIndex -= dir) {
         moved = 0;
         do { /// skip all hidden items
-            if (index + dir >= GetCount() || index + dir < 0) {
+            moved += dir;
+
+            if (IS_OUT_OF_RANGE(index + moved, 0, GetCount() - 1)) {
                 /// cursor would get out of menu
+                moveIndex = 0;
                 return false;
             }
-            moved += dir;
+
             item = GetItem(index + moved);
-            if (item == nullptr)
+            if (item == nullptr) {
+                moveIndex = 0;
                 return false;
+            }
         } while (item->IsHidden());
-        SetIndex(index + moved); /// sets new cursor position (visible item)
+        SetIndex(uint8_t(index + moved)); /// sets new cursor position to a visible item
```

Comment:
what about boundaries use `std::clamp` or `std::min` and `std::max` to check?

Reply:
There's a check on line 98.